src/sp/it/pl/util/file/Util.kt

```diff
@@ -13,53 +13,49 @@ import kotlin.streams.asStream
- *
- * Use instead of [File.getName] which returns empty string for root directories.
+ * Returns the [File.getName] or [File.toString] if this is the root directory,
+ * since that returns an empty string otherwise.
  *
  * @return name of the file with suffix
  */
-val File.nameOrRoot: String get() = name.takeUnless { it.isEmpty() } ?: toString()
+val File.nameOrRoot: String
+    get() = name.takeUnless { it.isEmpty() } ?: toString()
 
 /**
- * For files name with no extension is returned.
- * For directories name is returned.
- * Root directory returns 'X:\' string.
+ * Returns the [File.nameWithoutExtension] or [File.nameOrRoot] if this is a directory.
  *
- * @return name of the file without suffix
+ * @return name of the file without extension
  */
-// TODO: does not work for directories with '.'
-val File.nameWithoutExtensionOrRoot: String get() = nameWithoutExtension.takeUnless { it.isEmpty() } ?: toString()
+val File.nameWithoutExtensionOrRoot: String
+    get() = if(isDirectory) nameWithoutExtension else nameOrRoot
 
-/** @returns file itself if exists or its first existing parent or error if null or no parent exists */
+/** @return file itself if exists or its first existing parent or error if no parent exists */
 fun File.find1stExistingParentFile(): Try<File, Void> = when {
     exists() -> Try.ok(this)
     else -> parentDir?.find1stExistingParentFile() ?: Try.error()
 }
 
-/** @returns file's first existing parent or error if null or no parent exists */
+/** @return first existing directory in this file's hierarchy or error if no parent exists */
 fun File.find1stExistingParentDir(): Try<File, Void> = when {
     exists() && isDirectory -> Try.ok(this)
     else -> parentDir?.find1stExistingParentDir() ?: Try.error()
 }
 
-fun File.childOf(childName: String) = File(this, childName)
+inline fun File.childOf(childName: String) = File(this, childName)
 
-fun File.childOf(childName: String, childName2: String) = childOf(childName).childOf(childName2)
+inline fun File.childOf(childName: String, childName2: String) = childOf(childName).childOf(childName2)
 
-fun File.childOf(childName: String, childName2: String, childName3: String) = childOf(childName, childName2).childOf(childName3)
+inline fun File.childOf(childName: String, childName2: String, childName3: String) = childOf(childName, childName2).childOf(childName3)
 
-fun File.childOf(vararg childNames: String) = childNames.fold(this, File::childOf)
+inline fun File.childOf(vararg childNames: String) = childNames.fold(this, File::childOf)
 
-fun File.isParentOf(child: File) = child.parentDir==this
+inline infix fun File.isChildOf(parent: File) = parent.isParentOf(this)
 
-fun File.isAnyParentOf(child: File) = generateSequence(child, { it.parentDir }).any { isParentOf(it) }
+inline fun File.isAnyChildOf(parent: File) = parent.isAnyParentOf(this)
 
-fun File.isChildOf(parent: File) = parent.isParentOf(this)
+inline infix fun File.isParentOf(child: File) = child.parentDir==this
```

Comment:
inability to chain the call? You can still use it in non-infix notation as well, so no features are removed. Why would I refactor all the existing usages? It's just written there as a possibility to be used in the future in infix.

Reply:
You are right, I mistakenly did not even think about using an infix function any other way. Anyway, unless majority of the usages should be infix I would advise against infix support. If you believe this case can leverage from infix mark it as resolved. I remain 50:50.

Reply:
`file isParentOf otherFile` feels pretty natural to me, so that's why I want to keep the possibility open.  Having it there does not force anyone to use it neither harm anything, so why not have it?